### PR TITLE
Add/filter params to queries

### PR DESCRIPTION
**Pull request by psealock**
Create a function `getFilterQuery` to gather results from filters, comparisons, and advancedFilters to apply correct query parameters on data requests.

### Simple Filters

These will update the tables correctly, but I think it would be nice to update the tab on the chart (in a separate issue).

screen shot 2018-10-24 at 10 42 23 pm

1. `/wp-admin/admin.php?page=wc-admin#/analytics/products?filter=top_sales`.
2. The query should include `orderBy: 'gross_revenue'`,  `order: 'desc'`.
3. Repeat for "Top Products By Items Sold".

### Compare Filters

screen shot 2018-10-24 at 10 50 06 pm

1.` /wp-admin/admin.php?page=wc-admin#/analytics/products?filter=compare-product`.
2. Choose at least two products.
3. Click "Compare".
4. See `products: 64,66` included in the requests.
5. The table and chart data will reflect the filter.
6. Repeat for comparing categories.

### Single Filters

screen shot 2018-10-24 at 10 52 06 pm

1.` /wp-admin/admin.php?page=wc-admin#/analytics/products?filter=compare-product`.
2. Choose a product.
4. See `products: 66` included in the requests and see the Table and Chart results filtered accordingly.

### Advanced Filters

screen shot 2018-10-24 at 10 54 06 pm

_**Note:** Many of the Advanced Filters parameters have not yet been implemented on the back end and will not actually filter results_. Testing is best achieved through the Network tab and making sure requests to `/reports/orders` and `/reports/orders/stats` have the correct parameters added.

screen shot 2018-10-24 at 10 48 16 pm

1. `/wp-admin/admin.php?page=wc-admin#/analytics/orders?filter=advanced`.
2. Select filters and add items to filter.
3. See the parameters included in requests. These will be the same ones found in the url.
4. Make sure the `match` parameter is passed as well.

### Additional things to consider

* Query parameters having nothing to do with filters should not be included in requests
* Back end APIs aren't finished, so the exact terms may change
* If you're reading this far, thank you!

### CODE REVIEW

File: client/components/filters/filter/index.js [hunk 1]
```diff
@@ -16,6 +16,7 @@ import AnimationSlider from 'components/animation-slider';
 import DropdownButton from 'components/dropdown-button';
 import Search from 'components/search';
 import { updateQueryString } from 'lib/nav-utils';
+import { flatenFilters } from './utils';
```
Comment (belcherj) [inline client/components/filters/filter/index.js]: Nice!

**Review by belcherj — APPROVED** [review]
Tested and works as expected.  Still lots to do on the front end to make all this work.  The charts and summaries are not displaying correctly for the filtered data.

**Comment by LevinMedia** [comment]
@psealock one small fix - can you update the last filter to read "Top Products by Gross Revenue" 

> These (top items sold / gross revenue )  will update the tables correctly, but I think it would be nice to update the tab on the chart (in a separate issue).

That would be a nice touch. There's some additional behavior for how this will manifest itself in the chart and tabular data as well - The short of it is, clicking on either of the top products filters will automatically create a  comparison of the top products (ordered by items sold, or gross revenue) The chart should change to a comparison chart that has one extra feature, a setting to let you control how many products to include in the comparison. 

screen shot 2018-10-24 at 11 48 19 am

The tabular data will only display the selected number of top products, and will be ordered by either items sold, or gross revenue, depending on the selected filter.  Lastly, the summary numbers will reflect just the data from the top products.

**Comment by LevinMedia** [comment]
> Compare Filters

Looking good! Just making note that the chart should also be switched out from the standard chart to the comparison chart when it's ready.

**Review by Aljullu — COMMENTED** [review]
This looks really great. Good job @psealock! :clap: 

I added some small comments, but it's looking good!

File: client/analytics/report/orders/config.js [hunk 2]
```diff
@@ -18,7 +18,7 @@ export const filters = [
 ];
 
 /*eslint-disable max-len*/
-export const advancedFilterConfig = {
+export const advancedFilters = {
```
[the remark this answers is not in the document]
Comment (ryelle) [reply client/analytics/report/orders/config.js]: I would suggest leaving the individual `const`s exported (`export const advancedFilters = …`), and when you want both in one object, you can do 

```
import * as ordersConfig from 'analytics/report/orders/config';
import * as productsConfig from 'analytics/report/products/config';
```

File: client/analytics/report/products/index.js [hunk 3]
```diff
@@ -55,6 +55,7 @@ export default compose(
 		const primaryData = getReportChartData( 'products', 'primary', query, select );
 
 		const { getProducts, isGetProductsError, isGetProductsRequesting } = select( 'wc-admin' );
+		const filterQuery = getFilterQuery( 'products', query );
 		const tableQuery = {
```
[the remark this answers is not in the document]
Comment (psealock) [reply client/analytics/report/products/index.js]: @Aljullu what do you think about using `withSelect` in `ProductsReportTable` and moving all the relevant parts such as `isProductsError` etc? We wouldn't have to pass both query and tableQuery and there would be more clarity around `isProductsError` only referring to the request for table data, eg `/reports/products` not `/reports/products/stats`.

**Comment by psealock** [comment]
Thank you @Aljullu @ryelle for the comments. I've addressed them including the following fixes:

* Removed default export from configs.
* Added `tableQuery` prop to `ProductsReportTable`.

**Review by Aljullu — APPROVED** [review]
Everything looks good to me! :rocket:

File: client/analytics/report/products/index.js (hunk 3 above)
[the remark this answers is not in the document]
Comment (Aljullu) [reply client/analytics/report/products/index.js]: > what do you think about using `withSelect` in `ProductsReportTable`

I created an issue for that and assigned it to myself: #703.

**Pull request merged**